### PR TITLE
docs: form

### LINKED ISSUE #831: docs: ui.form

### PR DESCRIPTION
**Pull request by ethanalvizo**
Closes #831

### CODE REVIEW

**Comment by dsmmcken** [comment]
testing blocked by #948

File: plugins/ui/docs/components/form.md [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+
+```python
+from deephaven import ui
+
+
+@ui.component
+def ui_form():
+    return ui.form(
+        ui.text_field(name="name", label="Enter name"),
+        ui.button("Submit", type="submit"),
```
Comment (dsmmcken) [inline plugins/ui/docs/components/form.md:15]:
```suggestion
        ui.button("Submit", type="submit"),
```

We probably want to somehow do nothing with this submit event, not sure how though. @mofojed ?

While some forms will probably be traditional and go to an external end point, I imagine most would be back to dh.ui. So we should show how to handle that in more of the examples.

File: plugins/ui/docs/components/form.md [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+def ui_form_emphasized():
+
+    return ui.form(
+        ui.text_field(name="name", label="Enter name"),
+        is_emphasized=True,
+    )
+
+
+my_form = ui_form_emphasized()
+```
```
Comment (dsmmcken) [inline plugins/ui/docs/components/form.md:104]: for quiet and emphasized, show a few more component types mixed in so the output actually shows what were are talking about.

Emphasized has no effect on text_fields, so the example shows nothing. Needs other components.

**Review by dsmmcken — CHANGES_REQUESTED** [review]

File: plugins/ui/docs/components/form.md (hunk 1 above)
Comment (mofojed) [reply plugins/ui/docs/components/form.md:15]: Yes we should use the `on_submit` event. Could just be something like:
```python
from deephaven import ui


@ui.component
def ui_form():
    return ui.form(
        ui.text_field(name="name", label="Enter name"),
        ui.button("Submit", type="submit"),
        on_submit=lambda e: print(f"Form submitted: {e}")
    )

my_form = ui_form()
```

We should also have examples using the `on_reset` and `on_invalid` events as well.

File: plugins/ui/docs/components/form.md (hunk 2 above)
Comment (ethanalvizo) [reply plugins/ui/docs/components/form.md:104]: image
[The spectrum docs](https://react-spectrum.adobe.com/react-spectrum/Form.html#emphasized) are pretty funny for this example too. Idk if it's just my eyes but I don't think emphasized does anything for RadioField either

Comment (dsmmcken) [reply plugins/ui/docs/components/form.md:175]: Fills the circle with the accent color instead of gray.

**Review by mofojed — APPROVED** [review]

File: plugins/ui/docs/components/form.md [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+@ui.component
+def ui_form_submit():
+    return ui.form(
+        ui.text_field(name="name", label="Enter name"),
+        ui.button("Reset", type="reset"),
+        on_reset=lambda e: print(f"Form reset"),
+    )
+
+
+my_form_submit = ui_form_submit()
```
Comment (dsmmcken) [inline plugins/ui/docs/components/form.md:64]: Should show with more than one component in the form.

**Review by dsmmcken — DISMISSED** [review]

File: plugins/ui/docs/components/form.md [hunk 4]
```diff
@@ -0,0 +1,261 @@
+# Form
+
+Forms allow users to enter data that can be submitted while providing alignment and styling for form fields
```
Comment (margaretkennedy) [inline plugins/ui/docs/components/form.md:3]:
```suggestion
Forms allow users to enter data that can be submitted while providing alignment and styling for form fields.
```

File: plugins/ui/docs/components/form.md [hunk 5]
```diff
@@ -0,0 +1,261 @@
...
+        target="_blank",
+    )
+
+
+my_form_action = ui_form_action()
+```
+
+## Validation Behavior
+
+By default, validation errors will be displayed in real-time as the fields is edited but form submission is not blocked. To enable this and native HTML form validation, set `validation_behavior` to "native"
```
Comment (margaretkennedy) [inline plugins/ui/docs/components/form.md:121]:
```suggestion
By default, validation errors will be displayed in real-time as the fields are edited, but form submission is not blocked. To enable this and native HTML form validation, set `validation_behavior` to "native".
```

File: plugins/ui/docs/components/form.md [hunk 6]
```diff
@@ -0,0 +1,261 @@
...
+        is_disabled=True,
+    )
+
+
+my_form_disabled = ui_form_disabled()
+```
+
+## Necessity Indicator
+
+The `necessity_indicator` prop dictates whether form labels will use an icon or a label to outline which form fields are required. The default is "icon"
```
Comment (margaretkennedy) [inline plugins/ui/docs/components/form.md:208]:
```suggestion
The `necessity_indicator` prop dictates whether form labels will use an icon or a label to outline which form fields are required. The default is "icon".
```

**Review by dsmmcken — APPROVED** [review]

**Review by margaretkennedy — APPROVED** [review]

**Pull request merged**